### PR TITLE
PSS signatures

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -2201,9 +2204,13 @@ hash
...
-  {{RFC3447}}, DSA {{DSS}}, and ECDSA {{ECDSA}}, respectively.  The
-  "anonymous" value is meaningless in this context but used in
+  The values indicate anonymous signatures, RSASSA-PKCS-v1_5,
+  {{RFC3447}}, DSA {{DSS}}, ECDSA {{ECDSA}}, and
+  RSASSA-PSS respectively. Because all RSA signatures used in the
+  TLS protocol itself (as opposed to those in certificates)
+  are RSASSA-PSS, the "rsa" value refers solely to signatures
+  which appear in certificates.
+  The "anonymous" value is meaningless in this context but used in
   {{server-key-share}}.  It MUST NOT appear in this extension.
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: BTW, The use of the "anonymous" value isn't actually defined in the server-key-share section.

Comment (beurdouche) [inline draft-ietf-tls-tls13.md]: In the “Remove DSA signature” PR (#235), I added deprecation for the “anonymous” value based on Dave’s advice…

Comment (ekr) [inline draft-ietf-tls-tls13.md]: I'm still waiting for the chairs to declare consensus on PR#235.

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -2201,9 +2204,13 @@ hash
 
 signature
 : This field indicates the signature algorithm that may be used.
-  The values indicate anonymous signatures, RSASSA-PKCS1-v1_5
-  {{RFC3447}}, DSA {{DSS}}, and ECDSA {{ECDSA}}, respectively.  The
-  "anonymous" value is meaningless in this context but used in
+  The values indicate anonymous signatures, RSASSA-PKCS1-v1_5,
+  {{RFC3447}}, DSA {{DSS}}, ECDSA {{ECDSA}}, and
+  RSASSA-PSS respectively. Because all RSA signatures used in the
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: Please add a reference link for this, as well.

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -2183,7 +2186,7 @@ The "extension_data" field of this extension contains a
 
 Each SignatureAndHashAlgorithm value lists a single hash/signature pair that
 the client is willing to verify. The values are indicated in descending order
-of preference.
+of preference.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: nit: diff shows an added trailing space

File: draft-ietf-tls-tls13.md [hunk 4]
```diff
@@ -2201,9 +2204,13 @@ hash
 
 signature
 : This field indicates the signature algorithm that may be used.
-  The values indicate anonymous signatures, RSASSA-PKCS1-v1_5
-  {{RFC3447}}, DSA {{DSS}}, and ECDSA {{ECDSA}}, respectively.  The
-  "anonymous" value is meaningless in this context but used in
+  The values indicate anonymous signatures, RSASSA-PKCS1-v1_5,
+  {{RFC3447}}, DSA {{DSS}}, ECDSA {{ECDSA}}, and
+  RSASSA-PSS respectively. Because all RSA signatures used in the
+  TLS protocol itself (as opposed to those in certificates)
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: The "in the TLS protocol itself" is kinda vague and rather imprecise. Just be explicit and say "in signed TLS handshake messages (see {{digital-signing}})".

File: draft-ietf-tls-tls13.md [hunk 5]
```diff
@@ -313,6 +313,11 @@ server: The endpoint which did not initiate the TLS connection.
 ##  Major Differences from TLS 1.2
 
 
+draft-09
+
+- Change to RSA-PSS signatures.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: Just to be clear what is changed:
"signatures" -> "signatures for handshake messages"

File: draft-ietf-tls-tls13.md (hunk 2 above)
Comment (tomato42) [inline draft-ietf-tls-tls13.md]: I'd also add a note for IANA to rechristen the "RSA" in TLS SignatureAlgorithm Regristry to "RSASSA-PKCS1-v1_5" or something equally unambiguous.

**Pull request merged**